diff --git a/node_modules/prosemirror-view/dist/index.cjs b/node_modules/prosemirror-view/dist/index.cjs
--- a/node_modules/prosemirror-view/dist/index.cjs
+++ b/node_modules/prosemirror-view/dist/index.cjs
@@ -3058,6 +3058,7 @@ function selectClickedNode(view, inside) {
 }
 function updateSelection(view, selection, origin) {
   if (!view.focused) view.focus();
+  if (selection.$from.doc != view.state.doc) selection = prosemirrorState.Selection.near(view.state.doc.resolve(Math.min(selection.from, view.state.doc.content.size)));
   if (view.state.selection.eq(selection)) return;
   var tr = view.state.tr.setSelection(selection);
   if (origin == "pointer") tr.setMeta("pointer", true);
diff --git a/node_modules/prosemirror-view/dist/index.js b/node_modules/prosemirror-view/dist/index.js
--- a/node_modules/prosemirror-view/dist/index.js
+++ b/node_modules/prosemirror-view/dist/index.js
@@ -3214,6 +3214,8 @@ function runHandlerOnContext(view, propName, pos, inside, event) {
 function updateSelection(view, selection, origin) {
     if (!view.focused)
         view.focus();
+    if (selection.$from.doc != view.state.doc)
+        selection = Selection.near(view.state.doc.resolve(Math.min(selection.from, view.state.doc.content.size)));
     if (view.state.selection.eq(selection))
         return;
     let tr = view.state.tr.setSelection(selection);
diff --git a/node_modules/prosemirror-view/src/input.ts b/node_modules/prosemirror-view/src/input.ts
--- a/node_modules/prosemirror-view/src/input.ts
+++ b/node_modules/prosemirror-view/src/input.ts
@@ -186,5 +186,7 @@ function updateSelection(view: EditorView, selection: Selection, origin: string) {
 function updateSelection(view: EditorView, selection: Selection, origin: string) {
   if (!view.focused) view.focus()
+  if (selection.$from.doc != view.state.doc)
+    selection = Selection.near(view.state.doc.resolve(Math.min(selection.from, view.state.doc.content.size)))
   if (view.state.selection.eq(selection)) return
   let tr = view.state.tr.setSelection(selection)
   if (origin == "pointer") tr.setMeta("pointer", true)
